javascript/packages/orchestrator/src/providers/podman/resources/node.resource.ts

```diff
@@ -0,0 +1,177 @@
+import { getRandomPort, makeDir } from "@zombienet/utils";
+import { genCmd, genCumulusCollatorCmd } from "../../../cmdGenerator";
+import {
+  P2P_PORT,
+  PROMETHEUS_PORT,
+  RPC_HTTP_PORT,
+  RPC_WS_PORT,
+} from "../../../constants";
+import { Node } from "../../../types";
+import { Client } from "../../client";
+
+export class NodeResource {
+  private readonly configPath: string;
+  private readonly dataPath: string;
+  private readonly relayDataPath: string;
+
+  constructor(
+    client: Client,
+    protected readonly namespace: string,
+    protected readonly nodeSetupConfig: Node,
+  ) {
+    const nodeName = nodeSetupConfig.name;
+    this.configPath = `${client.tmpDir}/${nodeName}/cfg`;
+    this.dataPath = `${client.tmpDir}/${nodeName}/data`;
+    this.relayDataPath = `${client.tmpDir}/${nodeName}/relay-data`;
+  }
+
+  public async generateSpec() {
+    const volumes = await this.generateVolumes();
+    const volumeMounts = this.generateVolumesMounts();
+    const containersPorts = await this.generateContainersPorts();
+    const containers = await this.generateContainers(
+      volumeMounts,
+      containersPorts,
+    );
+    const podManifest = this.generatePodSpec(containers, volumes);
+
+    return podManifest;
+  }
+
+  private async createVolumeDirectories() {
+    await makeDir(this.configPath, true);
+    await makeDir(this.dataPath, true);
+    await makeDir(this.relayDataPath, true);
+  }
+
+  private async generateVolumes() {
+    await this.createVolumeDirectories();
+
+    const configVolume = {
+      name: "tmp-cfg",
+      hostPath: { type: "Directory", path: this.configPath },
+    };
+    const dataVolume = {
+      name: "tmp-data",
+      hostPath: { type: "Directory", path: this.dataPath },
+    };
+    const relayDataVolume = {
+      name: "tmp-relay-data",
+      hostPath: { type: "Directory", path: this.relayDataPath },
+    };
+
+    return [configVolume, dataVolume, relayDataVolume];
```

Comment:
Sorry for repeating that - IMO its better to avoid creating new vars if they are not reusable - just do the return at once; You can repeat wherever needed
```suggestion
    return [{
      name: "tmp-cfg",
      hostPath: { type: "Directory", path: this.configPath },
    }, {
      name: "tmp-data",
      hostPath: { type: "Directory", path: this.dataPath },
    }, {
      name: "tmp-relay-data",
      hostPath: { type: "Directory", path: this.relayDataPath },
    };];
```

Reply:
Yes, I'm use to it to make code cleaner with a quick glimpse, question of taste en colors, gonna fix it